docs/concepts/unions.md

```diff
@@ -285,10 +288,18 @@ class DiscriminatedModel(BaseModel):
 
 data = {'x': {'x': {'x': 'a'}}}
 m = DiscriminatedModel.model_validate(data)
-assert m == (
-    DiscriminatedModel(x=DiscriminatedModel(x=DiscriminatedModel(x='a')))
-)
-assert m.model_dump() == data
+print(m)
+#> x=DiscriminatedModel(x=DiscriminatedModel(x='a'))
+
+try:
```

Comment:
this example, together with an example or the errors raised without a custom error need to be in a separate example.